lib/cocoapods/downloader.rb

```diff
@@ -33,11 +33,16 @@ def self.download(
       cache_path: Config.instance.cache_root + 'Pods'
     )
       can_cache &&= !Config.instance.skip_download_cache
+
+      request = preprocess_request(request)
+
       if can_cache
         raise ArgumentError, 'Must provide a `cache_path` when caching.' unless cache_path
         cache = Cache.new(cache_path)
         result = cache.download_pod(request)
       else
+        raise ArgumentError, 'Must provide a `target` when caching is disabled.' unless target
```

Comment:
@segiddins this is a bit out of scope in this PR but I stumbled across it writing the test. Does this seem right to you?